### PR TITLE
rmd-22906 Refactor uploadFile tasks

### PR DESCRIPTION
**Pull request by DavidGamba**
Move logic to the Domain Object

@tonytvo please review.

### CODE REVIEW

File: src/main/groovy/com/calgaryscientific/gradle/VeracodeUploadFile.groovy [hunk 1]
```diff
@@ -30,6 +30,26 @@ import groovy.transform.CompileStatic
 
 @CompileStatic
 class VeracodeUploadFile {
+    static void uploadFiles(VeracodeAPI veracodeAPI, File outputFile, Set<File> fileSet, Integer maxTries, Integer waitTime, Boolean delete) {
```
Comment (tonytvo) [inline src/main/groovy/com/calgaryscientific/gradle/VeracodeUploadFile.groovy:33]: do we have test that check if the file is deleted?

File: src/main/groovy/com/calgaryscientific/gradle/VeracodeUploadFileSandboxTask.groovy [hunk 2]
```diff
@@ -63,13 +63,11 @@ class VeracodeUploadFileSandboxTask extends VeracodeTask {
     void run() {
         Integer maxTries = Integer.parseInt((this.maxUploadAttempts != null) ? this.maxUploadAttempts : '10')
         Integer waitTime = Integer.parseInt((this.waitTimeBetweenAttempts != null) ? this.waitTimeBetweenAttempts : '5000')
-        getFileSet().each { file ->
-            VeracodeUploadFile.uploadFile(file, maxTries, waitTime, veracodeAPI, getOutputFile(), true)
-            if (delete == "true") {
-                println("Deleting ${file}")
-                file.delete()
-            }
+        Boolean deleteBool = false
```
Comment (tonytvo) [inline src/main/groovy/com/calgaryscientific/gradle/VeracodeUploadFileSandboxTask.groovy]: I think you can just use .toBoolean() per this: http://mrhaki.blogspot.ca/2009/11/groovy-goodness-convert-string-to.html

File: src/main/groovy/com/calgaryscientific/gradle/VeracodeUploadFile.groovy [hunk 3]
```diff
@@ -30,6 +30,26 @@ import groovy.transform.CompileStatic
...
+            uploadFile(file, maxTries, waitTime, veracodeAPI, outputFile, false)
+            if (delete) {
+                println("Deleting ${file}")
+                file.delete()
+            }
+        }
+    }
+
+    static void uploadSandboxFiles(VeracodeAPI veracodeAPI, File outputFile, Set<File> fileSet, Integer maxTries, Integer waitTime, Boolean delete) {
+        fileSet.each { file ->
```
Comment (tonytvo) [inline src/main/groovy/com/calgaryscientific/gradle/VeracodeUploadFile.groovy:44]: I'm not comfortable with having too many responsibilities in a single loop, one way to tackle this is:
```
uploadSandboxFiles()
deleteFileIfConfigured()
```

Comment (DavidGamba) [reply src/main/groovy/com/calgaryscientific/gradle/VeracodeUploadFile.groovy:44]: I think that in this case creating a method for it is going too far. The uploadFile is already a method call.

```
            uploadFile(file, maxTries, waitTime, veracodeAPI, outputFile, true)
            if (delete) {
                println("Deleting ${file}")
                file.delete()
            }
```

vs

```
            uploadFile(file, maxTries, waitTime, veracodeAPI, outputFile, true)
            deleteFileIfConfigured(file, delete)
```

File: src/main/groovy/com/calgaryscientific/gradle/VeracodeUploadFile.groovy (hunk 1 above)
Comment (DavidGamba) [reply src/main/groovy/com/calgaryscientific/gradle/VeracodeUploadFile.groovy:33]: Yes, for both app level and sandbox level task. For example: https://github.com/calgaryscientific/veracode-gradle-plugin/blob/dd14c94995fd673b40e1209344c1553b96f83f9b/src/test/groovy/com/calgaryscientific/gradle/VeracodeUploadFileTest.groovy#L64

**Review by tonytvo — APPROVED** [review]

**Pull request merged**